src/components/button/button.ts

```diff
@@ -100,6 +100,26 @@ export class Button extends LitElement {
   @property({ type: String })
   name = '';
 
+  /** Determines if the button is Floatable */
+  @property({ type: Boolean })
+  isFloating = false;
+
+  /** Show button after scrolling to 50% of the page*/
+  @property({ type: Boolean })
+  showOnScroll = false;
+
+  /** Determines showButton state .
+   * @internal
+   */
+  @state()
+  showButton = false;
```

Comment:
Prefix internal/private variables & functions with an underscore `_`. Helps for readability.

Reply:
Addressed.